Content.Server/Atmos/GasVaporComponent.cs

```diff
@@ -0,0 +1,122 @@
+﻿using Content.Shared.Physics;
+using Content.Server.Atmos.Reactions;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.Components;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Interfaces.Map;
+using Robust.Shared.IoC;
+using Robust.Shared.Maths;
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+using Content.Server.GameObjects.Components.Atmos;
+using Content.Server.Interfaces;
+using Content.Shared.Atmos;
+
+namespace Content.Server.Atmos
+{
+    [RegisterComponent]
+    class GasVaporComponent : Component, ICollideBehavior, IGasMixtureHolder
+    {
+#pragma warning disable 649
+        [Dependency] private readonly IMapManager _mapManager = default!;
+#pragma warning enable 649
+        public override string Name => "GasVapor";
+
+        [ViewVariables] public GasMixture Air { get; set; }
+
+        [ViewVariables] private GridAtmosphereComponent _gridAtmosphereComponent;
+
+        private bool _running;
+        private Vector2 _direction;
+        private float _velocity;
+        private float _disspateTimer = 0;
+        private float _dissipationInterval;
+        private Gas _gas;
+        private float _gasVolume;
+        private float _gasTemperature;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            Air = new GasMixture(_gasVolume){Temperature = _gasTemperature};
+            Air.SetMoles(_gas,20);
```

Comment:
Maybe put the starting amount in the yaml? Or calculate it someway from the fluid that was taken from the fire extinguisher